### PR TITLE
Refactor DeckPickerContextMenu and it's related tests

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Refactors DeckPickerContextMenu :
- use newer apis when interacting with DeckPicker
- use AlertDialog instead of MaterialDialog(see below the before and after for the dialog change)
- add more tests to enforce the structure of the dialog/presence of dynamic options
- refactors old tests to optimize them and hopefully improve performance a bit

Note: while this looks like a lot of changes, most of the changes are moving code around. Commits should be reviewed one by one, each one has a specific purpose and build on the previous one.

## How Has This Been Tested?

Ran the tests, manually verified the dialog.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/dialogs/DeckPickerContextMenuTest.kt [hunk 1]
```diff
@@ -56,11 +65,141 @@ class DeckPickerContextMenuTest : RobolectricTest() {
...
+
+    @Test
+    fun `Shows standard options`() {
+        launch(
+            DeckPickerContextMenu::class.java,
+            withArguments(),
+            R.style.Theme_Light
+        ).onFragment { fragment ->
+            assertTrue(
+                fragment.foundOptions().contains(fragment.getString(R.string.menu_add)),
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/dialogs/DeckPickerContextMenuTest.kt:92]: nit: extract to a local `fun` - repetitive long line

**Review by david-allison — APPROVED** [review]
This is great!

Might want to use it as an opportunity to start moving logic to a ViewModel, but I'm happy as-is. Great forward-progress

**Comment by lukstbit** [comment]
Thought about using a ViewModel but decided not to do it here to not complicate the PR.
I've implemented the requested change, note that I combined the last two commits.

**Review by mikehardy — APPROVED** [review]
Looks great, confirmed working with a local run through of deck options stuff, including activity recreation (via phone orientation switches). Works well +1

**Pull request merged**